Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -607,7 +607,7 @@ bool RenderStyle::isIdempotentTextAutosizingCandidate(AutosizeStatus status) con
 
     if (fields.contains(AutosizeStatus::Fields::FixedHeight)) {
         if (fields.contains(AutosizeStatus::Fields::FixedWidth)) {
-            if (whiteSpace() == WhiteSpace::NoWrap) {
+            if (whiteSpaceCollapse() == WhiteSpaceCollapse::Collapse && textWrapMode() == TextWrapMode::NoWrap) {
```

Comment:
I don't think the collapsing of whitespace characters is particularly relevant here (maybe of newlines even).

maybe this should be: 

```suggestion
            if (!preserveNewline() && textWrapMode() == TextWrapMode::NoWrap) {
```

Only really makes a difference with `preserve-spaces`, but I think this expresses intent better (and also would handle the unimplemented `discard` value correctly).

I'm not sure if the collapsing of newlines matters here even, but worth checking with @whsieh

Reply:
I think this would be good to land in a separate PR. We'll be able to add a meaningful PR title and make it easier to revert/debug in the future.